### PR TITLE
feat(connection): attempt both ipv6 and ipv4 when no family entered

### PR DESCRIPTION
**Pull request by daprahamian**
If the user does not specify an IP family, we will attempt to
connect first via IPv6, and fall back to IPv4 if that fails

Currently WIP as I need to write unit tests for this.

### CODE REVIEW

**Comment by daprahamian** [comment]
Removed WIP, this should be ready for review

File: test/tests/functional/connection_tests.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+var bson = require('bson');
```
Comment (jlord) [inline test/tests/functional/connection_tests.js]: A lone `var`

File: lib/connection/connection.js [hunk 2]
```diff
@@ -528,100 +528,131 @@ function merge(options1, options2) {
...
-  _options = _options || {};
-  // Set the connections
-  if (connectionAccounting) addConnection(this.id, this);
-  // Check if we are overriding the promoteLongs
-  if (typeof _options.promoteLongs === 'boolean') {
-    self.responseOptions.promoteLongs = _options.promoteLongs;
-    self.responseOptions.promoteValues = _options.promoteValues;
-    self.responseOptions.promoteBuffers = _options.promoteBuffers;
+function makeSSLConnection(self, _options) {
+  var sslOptions = {
```
Comment (jlord) [inline lib/connection/connection.js]: Lone `var`

**Comment by daprahamian** [comment]
Updated to remove `var`s

**Pull request merged**